go_sequencer/common/l1tx.go

```diff
@@ -317,3 +317,87 @@ func L1CoordinatorTxFromBytes(b []byte, chainID *big.Int, tokamakAddress ethComm
 	}
 	return tx, nil
 }
+
+// BytesGeneric returns the generic representation of a L1Tx. This method is
+// used to compute the []byte representation of a L1UserTx, and also to compute
+// the L1TxData for the ZKInputs (at the HashGlobalInputs), using this method
+// for L1CoordinatorTxs & L1UserTxs (for the ZKInputs case).
+func (tx *L1Tx) BytesGeneric() ([]byte, error) {
+	var b [RollupConstL1UserTotalBytes]byte
+	copy(b[0:20], tx.FromEthAddr.Bytes())
+	if tx.FromBJJ != EmptyBJJComp {
+		pkCompL := tx.FromBJJ
+		pkCompB := SwapEndianness(pkCompL[:])
+		copy(b[20:52], pkCompB[:])
+	}
+	fromIdxBytes, err := tx.FromIdx.Bytes()
+	if err != nil {
+		return nil, Wrap(err)
+	}
+	copy(b[52:58], fromIdxBytes[:])
+
+	depositAmountFloat40, err := NewFloat40(tx.DepositAmount)
+	if err != nil {
+		return nil, Wrap(err)
+	}
+	depositAmountFloat40Bytes, err := depositAmountFloat40.Bytes()
+	if err != nil {
+		return nil, Wrap(err)
+	}
+	copy(b[58:63], depositAmountFloat40Bytes)
+
+	amountFloat40, err := NewFloat40(tx.Amount)
+	if err != nil {
+		return nil, Wrap(err)
+	}
+	amountFloat40Bytes, err := amountFloat40.Bytes()
+	if err != nil {
+		return nil, Wrap(err)
+	}
+	copy(b[63:68], amountFloat40Bytes)
+
+	//TODO: Check this with the smart contract team, If we'll get TokenID in the transactions. And if we need to handle that.
+	//Else we can update this for better memory management.
+	// copy(b[68:72], tx.TokenID.Bytes())
```

Comment:
I'm not sure simply commenting this out would work as expected, especially if this byte array is used for serialization or communication with other systems. We might run into data exchange or processing issues down the line. So my suggestion is to update `RollupConstL1UserTotalBytes` from 78 to 74 length and `copy(b[68:74], toIdxBytes[:])`. But if you think it's safe to simply comment out then we can keep it as it is 😄

Reply:
Yupp, Thanks @Intizar-T I'll update this.  I left this commented out as it was prior to our discussion, Where i was not sure if we'll keep the tokenID or not.

Reply:
@Shailu-s thanks for the detailed answer. Sounds good to me!